src/helpers/applicationMessageError.ts

```diff
@@ -12,8 +15,18 @@ export class ApplicationMesssageError extends Error {
 }
 
 // tslint:disable-next-line: max-classes-per-file
-export const makeApplicationError = (messageKey: string) => class extends ApplicationMesssageError {
+export const makeToastMessageError = (messageKey: string) => class extends ToastMesssageError {
   constructor(meta: ToastMeta) {
     super(messageKey, meta);
   }
 };
+
+/**
+ * Return @param error if it is instance of ApplicationError or @param customError in other cases
+ */
+export const makeApplicationError = (error: Error, customError: Error | (() => Error)) => {
```

Comment:
Naming is a little bit confusing. It's implying that this function creates instances of `class ApplicationError`, but it doesn't.
Hmm... maybe naming is as a file name: 'makeApplicationMessageError` is a good idea. It's just a suggestion.

Reply:
maybe something like `makeApplicationOrCustomError`?

Reply:
Sounds good :)

Reply:
in the case that the first argument is an applicationError, it doesn't actually make anything. maybe something like `ensureApplicationErrorType` or something. a verb like `ensure` or `default` or something that doesn't imply construction